### DISCUSSION TITLE
Widget System

### DISCUSSION THREAD

**Discussion by BlueFalconHD** [charmbracelet/gum#78] [Ideas]
A widget system would allow for making a bash file, for example, and treating that as something you could add to the main interface. Like if I had a file that added a filter for pizza toppings, and another one for sides, in 2 different files, widgets would allow you to merge both together side by side in the interface.

**Comment by maaslalani** [comment]
Hey, this is totally possible, just not with gum. You'll want to look at our [Bubble Tea](https://github.com/charmbracelet/bubbletea) and [Lip Gloss](https://github.com/charmbracelet/lipgloss) packages.

**Reply by BlueFalconHD** [reply]
I understand, but for someone who does not want to learn go just to make a very simple git tui, it would be great.

**Reply by mathaou** [reply]
> I understand, but for someone who does not want to learn go just to make a very simple git tui, it would be great.

I'm working on a way to define a custom gum program with a JSON file, to achieve essentially this.

**Reply by BlueFalconHD** [reply]
> > I understand, but for someone who does not want to learn go just to make a very simple git tui, it would be great.
> 
> I'm working on a way to define a custom gum program with a JSON file, to achieve essentially this.

This is awesome! :shipit:

**Comment by deicon** [comment]
Maybe [sunbeam](https://github.com/pomdtr/sunbeam) is what you might want to check out
Essentially building a Json to define the TUI and reusing any command available to produce the JSON/YAML and act on actions

See this example https://github.com/pomdtr/sunbeam/blob/main/examples/github/sunbeam.yml

**Reply by mathaou** [reply]
Literally exactly what I started but way better

So NVM I guess everyone just use that package looks dope
